src/set/mod.rs

```diff
@@ -0,0 +1,170 @@
+//! A concurrent hash set.
+//!
+//! See `HashSet` for details.
+
+use std::borrow::Borrow;
+use std::hash::Hash;
+
+use crate::epoch::{self, Guard};
+use crate::iter::Keys;
+use crate::HashMap;
+
+/// A concurrent hash set implemented as a `FlurryHashMap` where the value is `()`.
+///
+/// # Examples
+///
+/// ```
+/// use flurry::HashSet;
+///
+/// // Initialize a new hash set.
+/// let books = HashSet::new();
+///
+/// // Add some books
+/// books.insert("Fight Club");
+/// books.insert("Three Men In A Raft");
+/// books.insert("The Book of Dust");
+/// books.insert("The Dry");
+///
+/// // Check for a specific one.
+/// if !books.contains(&"The Drunken Botanist") {
+///     println!("We don't have The Drunken Botanist.");
+/// }
+///
+/// // Remove a book.
+/// books.remove(&"Three Men In A Raft");
+///
+/// // Iterate over everything.
+/// let guard = flurry::epoch::pin();
+/// for book in books.iter(&guard) {
+///     println!("{}", book);
+/// }
+/// ```
+#[derive(Debug)]
+pub struct HashSet<T: 'static, S = crate::DefaultHashBuilder>
+where
+    T: Sync + Send + Clone + Hash + Eq,
+    S: std::hash::BuildHasher,
+{
+    map: HashMap<T, (), S>,
+}
+
+impl<T> HashSet<T, crate::DefaultHashBuilder>
```

Comment:
`DefaultHashBuilder` is correct for `fn new`, but the other methods should allow a generic `S: BuildHasher`. Consider the others too: `with_capacity` using the default, `with_hasher` and `with_capacity_and_hasher` using `S`.